src/main/java/bio/terra/workspace/app/controller/WorkspaceApiController.java

```diff
@@ -78,4 +81,18 @@ private AuthenticatedUserRequest getAuthenticatedInfo() {
         jobService.retrieveJobResult(id, Object.class, userReq);
     return new ResponseEntity<>(jobResultHolder.getResult(), jobResultHolder.getStatusCode());
   }
+
+  @Override
+  public ResponseEntity<DataReferenceList> enumerateReferences(
+      @PathVariable("id") String id,
+      @Valid @RequestParam(value = "offset", required = false, defaultValue = "0") Integer offset,
+      @Valid @RequestParam(value = "limit", required = false, defaultValue = "10") Integer limit,
+      @Valid @RequestParam(value = "filterControlled", required = false, defaultValue = "all")
+          FilterControlledEnum filterControlled) {
+    ControllerValidationUtils.ValidatePaginationParams(offset, limit);
+    DataReferenceList enumerateResult =
+        dataReferenceService.enumerateDataReferences(
+            id, offset, limit, filterControlled, getAuthenticatedInfo());
+    return ResponseEntity.ok(enumerateResult);
+  }
```

Comment:
honest question, I don't know the answer: is there any way to centrally define the offset and limit (and maybe "filter") `@RequestParam`s, and then have multiple methods inherit those definitions?

As we build out more APIs, I want to make sure that we are consistent in our pagination, sorting, and filtering parameters. Defining them centrally and inheriting them would be a good way to do that.

Also - probably want to add sorting parameters to this API.

Reply:
OpenAPI v3 has support for shared definitions. I have not used it yet, but it is one of the big benefits of moving from v2 to v3.

Reply:
I can move the parameters to a shared place in the OpenAPI doc, so we could generate all the parameters in methods of the WorkspaceApi interface with the same settings (names, types, defaults, etc). 
To be honest, I'm not totally sure how that affects our implementation of the interface - I don't really understand how annotations mix with interfaces well enough to say if that changes anything, since we're re-specifying the annotations manually.